src/coord/src/coord.rs

```diff
@@ -2951,15 +2963,14 @@ impl Coordinator {
 
                     // We want to prevent compaction of the indexes consulted by
                     // determine_timestamp, not the ones listed in the query.
-                    let (timestamp, timestamp_ids) =
+                    let determined =
                         self.determine_timestamp(session, &timedomain_ids, PeekWhen::Immediately)?;
                     // Add the used indexes to the recorded ids.
-                    timedomain_ids.extend(&timestamp_ids);
-                    // TODO: Delete the above once the below is confirmed to replace it.
+                    timedomain_ids.extend(&determined.compute_ids);
```

Comment:
I can't tell, but are we sure this shouldn't also include `determined.storage_ids`? My understanding from @mjibson is that this exists to ensure that indexes added in the transaction are captured too, but .. can we also add sources? I don't know.

Reply:
Yeah, adding the unmaterialized sources that were used in timestamp determination makes sense to me. I've updated the PR accordingly. Seems to pass locally. I'm expecting green across the board here.

Reply:
Done!